tests/unit/model/message_test.py

```diff
@@ -819,6 +819,23 @@ def test_undefined_variable(
     )
 
 
+def test_undefined_variable_boolean_condition_value() -> None:
+    mod_type = ModularInteger("P::MT", Pow(Number(2), Number(32)))
+    structure = [
+        Link(INITIAL, Field("F1")),
+        Link(Field("F1"), Field("F2"), condition=Variable("X", location=Location((10, 20)))),
+        Link(Field("F2"), FINAL),
+    ]
+    types = {Field("F1"): mod_type, Field("F2"): mod_type}
+
+    assert_message_model_error(
+        structure,
+        types,
+        r'^<stdin>:10:20: model: error: undefined variable "X"\n'
+        r"<stdin>:10:20: model: info: on path F1 -> F2$",
+    )
```

Comment:
How about also testing for undefined variables in size and first expressions?

Reply:
Fixed.